src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp

```diff
@@ -242,22 +242,38 @@ void CompileModelCacheTestBase::run() {
         GTEST_FAIL() << "Can't compile network without cache for " << m_functionName << " with precision " << m_precision.get_type_name() << std::endl;
     }
     auto originalOutputs = get_plugin_outputs();
-
+    size_t blobSizeInitial = -1;
+    size_t blobSizeAfterwards = -1;
     for (int i = 0; i < 2; i++) {
         // Step 2: Load with cache. Export or import shall not throw
         compiledModel = {}; // Destroy network object
         inferRequest = {};
         {
             core->set_property(ov::cache_dir(m_cacheFolderName));
             ASSERT_NO_THROW(compiledModel = core->compile_model(function, targetDevice, configuration));
-            if (targetDevice.find("AUTO") == std::string::npos)
+            if (targetDevice.find("AUTO") == std::string::npos) {
                 // Apply check only for HW plugins
                 ASSERT_EQ(i != 0, compiledModel.get_property(ov::loaded_from_cache));
+            } else {
+                while (true) {
+                    auto exeDevices = compiledModel.get_property(ov::execution_devices);
+                    if (exeDevices.size() == 1 && exeDevices.front().find("(CPU)") != std::string::npos)
+                        continue;
+                    break;
+                }
```

Comment:
the logic here seems not useful for the check below, no matter hardware is finished loading, you are not checking the exact blob number below